data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java

```diff
@@ -155,16 +174,31 @@ public KustoOperationResult executeStreamingIngest(String database, String table
         if (timeoutMs == null) {
             timeoutMs = STREAMING_INGEST_TIMEOUT_IN_MILLISECS;
         }
-        return Utils.post(clusterEndpoint, null, stream, timeoutMs.intValue() + CLIENT_SERVER_DELTA_IN_MILLISECS, headers, leaveOpen);
+        String response = Utils.post(clusterEndpoint, null, stream, timeoutMs.intValue() + CLIENT_SERVER_DELTA_IN_MILLISECS, headers, leaveOpen);
+        try {
+            return new KustoOperationResult(response, clusterEndpoint.endsWith("v2/rest/query") ? "v2" : "v1");
```

Comment:
This line has some logic in it, and appears both here and in execute. 
Maybe it should be a method, even a public facing one - just in case someone wants to turn their json into KustoOperationResult.

Reply:
Actually its always false here ..

Reply:
I really wanted to add a new KustoOperationResult constructor that accepts a URL so that this logic would be delegated to there. I think that conceptually makes most sense.
But I saw adding a new KustoOperationResult constructor as out of scope for this PR.

@ohadbitt if it's necessarily false, should I remove it altogether?

Reply:
Sorry I missed that, I see now that the cluster endpoint never ends in "v2/rest/query" and is always "v1".